implementations/micrometer-registry-timestream2/src/main/java/io/micrometer/timestream2/TimeStreamMeterRegistry.java

```diff
@@ -0,0 +1,343 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.exception.AbortedException;
+import software.amazon.awssdk.services.timestreamwrite.TimestreamWriteAsyncClient;
+import software.amazon.awssdk.services.timestreamwrite.model.*;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.StreamSupport.stream;
+import static software.amazon.awssdk.services.timestreamwrite.model.TimeUnit.SECONDS;
+
+/**
+ * {@link MeterRegistry} for TimeStream.
+ *
+ * @author Guillaume Hiron
+ * @since 1.6.0
+ */
+public class TimeStreamMeterRegistry extends StepMeterRegistry {
+
+    private static final ThreadFactory DEFAULT_THREAD_FACTORY = new NamedThreadFactory("timestream-metrics-publisher");
+
+    private static final WarnThenDebugLogger warnThenDebugLogger = new WarnThenDebugLogger(TimeStreamMeterRegistry.class);
+
+    private final Logger logger = LoggerFactory.getLogger(TimeStreamMeterRegistry.class);
+
+    private final TimeStreamConfig config;
+
+    private final TimestreamWriteAsyncClient timestreamWriteClient;
+
+    private volatile boolean checkedForTable = false;
+
+
+    /**
+     * Create a new instance with given parameters.
+     *
+     * @param config configuration to use
+     * @param clock  clock to use
+     * @param client TimeStream client to use
+     * @since 1.6.0
+     */
+    protected TimeStreamMeterRegistry(TimeStreamConfig config, Clock clock, TimestreamWriteAsyncClient client) {
+        this(config, clock, DEFAULT_THREAD_FACTORY, client);
+    }
+
+    /**
+     * Create a new instance with given parameters.
+     *
+     * @param config        configuration to use
+     * @param clock         clock to use
+     * @param threadFactory thread factory to use
+     * @param client        TimeStream client to use
+     * @since 1.6.0
+     */
+    protected TimeStreamMeterRegistry(TimeStreamConfig config, Clock clock, ThreadFactory threadFactory, TimestreamWriteAsyncClient client) {
+        super(config, clock);
+        config().namingConvention(new TimeStreamNamingConvention());
+        this.config = config;
+        checkClient(client);
+        this.timestreamWriteClient = client;
+        start(threadFactory);
+    }
+
+    public static Builder builder(TimeStreamConfig config) {
+        return new Builder(config);
+    }
+
+    private void checkClient(TimestreamWriteAsyncClient client) {
+        Objects.requireNonNull(client);
+    }
+
+    @Override
+    protected void publish() {
+        for (List<Meter> partition : MeterPartition.partition(this, config.batchSize())) {
+            sendRecords(metricData(partition));
+        }
+    }
+
+    //VisibleForTesting
+    List<Record> metricData(List<Meter> partition) {
```

Comment:
I would call it what it does: `toRecords`, in this case the call in `publish` will be a little cleaner too: `sendRecords(toRecords(partition));`. Also, if the loop above will ever be converted to a stream, this probably will be called from a `map`: `.map(this::toRecords)`.

Reply:
done